qiskit_machine_learning/algorithms/classifiers/vqc.py

```diff
@@ -154,9 +171,13 @@ def _fit_internal(self, X: np.ndarray, y: np.ndarray) -> OptimizerResult:
         """
         X, y = self._validate_input(X, y)
         num_classes = self._num_classes
-        cast(CircuitQNN, self._neural_network).set_interpret(
-            self._get_interpret(num_classes), num_classes
-        )
+
+        if isinstance(self._neural_network, SamplerQNN):
+            self._neural_network.set_interpret(self._get_interpret(num_classes), num_classes)
+        else:
```

Comment:
Perhaps, instead of `else` should be `elif`.

Reply:
sure